### ISSUE TITLE
Nov: DRC and o²S²PARC API Extension

### ISSUE THREAD

**Issue by elisabettai** [ITISFoundation/osparc-issues#1095]
### Description:

1. personal preferences, remaining simulation hours (for users with a limited resource budget)
2. define the hardware on which a user’s services shall run
3. passing environment variables for licensing, user information (such as email and tokens), and API keys for 3rd party services (such as Turn) to running services via the sidecar API
4.  real-time streaming of advanced progress information, such as solver convergence plots

### Deliverable:
Extended API functionality

### Acceptance criteria:
API functionality that permits to perform the above listed functions 

### Deadline:
Q1

[wrike Y7-MS 4.1: DRC and o²S²PARC API Extension](https://www.wrike.com/open.htm?id=1155714214)

**Comment by elisabettai** [comment]
In my understanding from the other cases the first three points above are done (1 and 3 as part of the dynamic sidecar work from @GitHK).

Is 2 done for computational and dynamic services as well? @mguidon, @sanderegg 

And 4 I think is still in progress, right @pcrespov?

**Comment by pcrespov** [comment]
> And 4 I think is still in progress, right @pcrespov?
- @elisabettai yes, this is still in progress. https://github.com/ITISFoundation/osparc-simcore/pull/4665

**Comment by mguidon** [comment]
- Number 2 is done for comp. services. Dynamics services are note instantiated via the API.
- Number 3: Missing is still the keys/secrets, the rest is done

**Comment by elisabettai** [comment]
@bisgaard-itis, @mguidon, job logs streaming is done, right? Should we close this one or there's still important bugs to be fixed before we can close it?

And a question: the job logs streaming can be used out-of-the-box with any comp. services (solvers)? Is it also available via the Python Client?

**Comment by bisgaard-itis** [comment]
> @bisgaard-itis, @mguidon, job logs streaming is done, right? Should we close this one or there's still important bugs to be fixed before we can close it?
> 
> And a question: the job logs streaming can be used out-of-the-box with any comp. services (solvers)? Is it also available via the Python Client?

I am fixing some stuff for the log streaming, but should be ready within the next couple of days. It will be made available from the python client, but isn't yet. I would say the other points are satisfied. I think that is basically the fact that the pricing plan and unit can be passed when running the job.

**Comment by pcrespov** [comment]
- @bisgaard-itis refers above to https://github.com/ITISFoundation/osparc-simcore/pull/5104

**Comment by elisabettai** [comment]
As far as I see, the missing PR for the log streaming has been merged.

We also reported that this milestone is completed in this way: 

"All the functionality listed in the description has been implemented and demonstrated at the monthly meeting on November 8th.
Personal preferences have been added to the dynamic backend to store global user preferences (platform-level) and preferences at the Services level. Support for environmental variables in the dynamic and computational backend have been added as well. Remaining simulation hours, in the form of credits can now be displayed to the users (only when the platform is used in a pay-per-use fashion). The API was also extended to support the creation of on-demand AWS machines with the user-defined specifications (amount of RAM, CPUs, GPUs). The logs of running computational jobs can now be shown in real time (thanks to the logs streaming). In this way, a user doesn’t need to wait for job termination before being able to access a log file.

In addition, we improved the API endpoints for file handling, making it easier to interact with files (e.g. file deletion). We improved performance, by only (re)saving files when their checksum has changed. 

We continued collaboration with and supporting MAP-Core, by releasing API entry points they require to query study inputs/outputs and we started establishing requirements for cross-core compatible json-schema support
"
For your info @pcrespov, @bisgaard-itis, @mguidon. I am closing this, let me know if you have any comments.
